### DISCUSSION TITLE
Adding `distinct_on` support for MySQL

### DISCUSSION THREAD

**Discussion by DanielBatesJ** [diesel-rs/diesel#2746] [Ideas]
Supporting `distinct_on` for MySQL database would be a useful feature to have. I'm not sure what the limitations are but I'm guessing it's related to a difference in syntax from PosgreSQL to MySQL. 

I did a quick check for any issues or requests mentioning this and didn't find anything, are there any PRs that look into this? And if not any advise for a beginner to contribute one?

Secondarily, the [docs](https://docs.diesel.rs/diesel/query_dsl/trait.QueryDsl.html#method.distinct_on) do not mention that the feature is only available on PosgreSQL, which lead to some confusion when writing my queries. Looking at the [source](https://docs.diesel.rs/src/diesel/query_dsl/mod.rs.html#166-172), however, the function is definitely labeled as a PosgreSQL specific feature. Is this something I should write up as a formal issue?

**Comment by weiznich** [comment]
Filling a feature request first is a good way to get some feedback on the idea before doing actual work. That's at least for me the preferred way to handle feature requests :+1:

Before starting to think about a potential implementation here are a few questions that we should answer first:

* Which versions of MySQL do support `DISTINCT ON` clauses? (Note: I did not find anything about support for `DISTINCT ON` clauses for MySQL in a quick search)
* What are the exact semantics of `DISTINCT ON` clauses for MySQL. Do they require similar restrictions like `DISTINCT ON` clauses for PostgreSQL. ([PostgreSQL documentation](https://www.postgresql.org/docs/current/sql-select.html#SQL-DISTINCT), did not find anything similar for MySQL)

If those questions are answered we can talk about the implementation. In fact if there is support in MySQL for the corresponding clause, it should be quite easy to add support to diesel.
